ddp-workspace/projects/ddp-pancan/src/app/components/footer/footer.component.ts

```diff
@@ -1,27 +1,23 @@
 import { Component, Inject } from '@angular/core';
 import { WindowRef } from 'ddp-sdk';
 import { ToolkitConfigurationService } from 'toolkit';
+import { BaseFooterComponent } from './base-footer/base-footer.component';
 
 @Component({
     selector: 'app-footer',
     templateUrl: './footer.component.html',
     styleUrls: ['./footer.component.scss']
 })
-export class FooterComponent {
+export class FooterComponent extends BaseFooterComponent {
     readonly phone: string;
     readonly email: string;
 
     constructor(
-        private windowRef: WindowRef,
-        @Inject('toolkit.toolkitConfig') config: ToolkitConfigurationService) {
-       this.phone = config.phone;
-       this.email = config.infoEmail;
-    }
-
-    public goToTop(): void {
-        this.windowRef.nativeWindow.scrollTo({
-            top: 0,
-            behavior: 'smooth'
-        });
+        private _windowRef: WindowRef,
+        @Inject('toolkit.toolkitConfig') config: ToolkitConfigurationService
+    ) {
+        super(_windowRef);
+        this.phone = config.phone;
+        this.email = config.infoEmail;
```

Comment:
how about making both footer components consistent and pass phone  and email via inputs here as well?

Reply:
Done.